src/command/Commands.ts

```diff
@@ -499,3 +512,338 @@ export class CommandLineSpec implements InputSpec<typeof StringT> {
 export function buildInputSpecByCommands(commands: Command[]): InputSpec<typeof StringT> | null {
   return new CommandLineSpec(commands);
 }
+
```

Comment:
Define command classes before all the build functions

Reply:
Moved.